RecoTracker/LSTCore/src/alpaka/PixelTriplet.h

```diff
@@ -0,0 +1,2789 @@
+    float x_OutLo = mdsInGPU.anchorX[thirdMDIndex];
+    float x_OutUp = mdsInGPU.anchorX[fourthMDIndex];
+
+    float y_InLo = mdsInGPU.anchorY[firstMDIndex];
+    float y_InUp = mdsInGPU.anchorY[secondMDIndex];
+    float y_OutLo = mdsInGPU.anchorY[thirdMDIndex];
+    float y_OutUp = mdsInGPU.anchorY[fourthMDIndex];
+
+    float rt_InOut = rt_InUp;
+
+    pass =
+        pass and (alpaka::math::abs(acc, SDL::deltaPhi(acc, x_InUp, y_InUp, x_OutLo, y_OutLo)) <= 0.5f * float(M_PI));
+    if (not pass)
+      return pass;
+
+    unsigned int pixelSegmentArrayIndex = innerSegmentIndex - rangesInGPU.segmentModuleIndices[pixelModuleIndex];
+    float ptIn = segmentsInGPU.ptIn[pixelSegmentArrayIndex];
+    float ptSLo = ptIn;
+    float px = segmentsInGPU.px[pixelSegmentArrayIndex];
+    float py = segmentsInGPU.py[pixelSegmentArrayIndex];
+    float pz = segmentsInGPU.pz[pixelSegmentArrayIndex];
+    float ptErr = segmentsInGPU.ptErr[pixelSegmentArrayIndex];
+    float etaErr = segmentsInGPU.etaErr[pixelSegmentArrayIndex];
+    ptSLo = alpaka::math::max(acc, ptCut, ptSLo - 10.0f * alpaka::math::max(acc, ptErr, 0.005f * ptSLo));
+    ptSLo = alpaka::math::min(acc, 10.0f, ptSLo);
+
+    float alpha1GeV_OutLo =
+        alpaka::math::asin(acc, alpaka::math::min(acc, rt_OutLo * k2Rinv1GeVf / ptCut, sinAlphaMax));
+    const float rtRatio_OutLoInOut =
+        rt_OutLo / rt_InOut;  // Outer segment beginning rt divided by inner segment beginning rt;
+
+    float dzDrtScale =
+        alpaka::math::tan(acc, alpha1GeV_OutLo) / alpha1GeV_OutLo;  // The track can bend in r-z plane slightly
+    const float zpitch_InLo = 0.05f;
+    const float zpitch_InOut = 0.05f;
+    float zpitch_OutLo = (isPS_OutLo ? pixelPSZpitch : strip2SZpitch);
+    float zGeom = zpitch_InLo + zpitch_OutLo;
+    zHi = z_InUp + (z_InUp + deltaZLum) * (rtRatio_OutLoInOut - 1.f) * (z_InUp < 0.f ? 1.f : dzDrtScale) +
+          (zpitch_InOut + zpitch_OutLo);
+    zLo = z_InUp + (z_InUp - deltaZLum) * (rtRatio_OutLoInOut - 1.f) * (z_InUp > 0.f ? 1.f : dzDrtScale) -
+          (zpitch_InOut + zpitch_OutLo);  //slope-correction only on outer end
+
+    pass = pass and ((z_OutLo >= zLo) && (z_OutLo <= zHi));
+    if (not pass)
+      return pass;
+
+    const float coshEta = alpaka::math::sqrt(acc, ptIn * ptIn + pz * pz) / ptIn;
+
+    const float drt_OutLo_InUp = (rt_OutLo - rt_InUp);
+
+    const float r3_InUp = alpaka::math::sqrt(acc, z_InUp * z_InUp + rt_InUp * rt_InUp);
+
+    float drt_InSeg = rt_InOut - rt_InLo;
+
+    const float sdlThetaMulsF = 0.015f * alpaka::math::sqrt(acc, 0.1f + 0.2f * (rt_OutLo - rt_InUp) / 50.f) *
+                                alpaka::math::sqrt(acc, r3_InUp / rt_InUp);
+    const float sdlMuls = sdlThetaMulsF * 3.f / ptCut * 4.f;  // will need a better guess than x4?
+
+    float dzErr = drt_OutLo_InUp * etaErr * coshEta;  //FIXME: check with the calc in the endcap
+    dzErr *= dzErr;
+    dzErr += 0.03f * 0.03f;  // pixel size x2. ... random for now
```

Comment:
Do we know what this `0.015` is? It seems to be the double of the pixel size? However, it doesn't correspond to the numbers that @YonsiG had sent:
> For 2S module, the stripe is 5cm * 90 μm.
For PS module pixel, 1.5mm * 100 μm.
For PS module stripe, 2.4cm * 100 μm.

Reply:
I think the number is related to the 1.5mm of pixel module size. And later, it says 0.03 is pixel size times 2 in the comments of L1367. However, unfortunately I'm not sure how this variable is scaled to 0.015..

Reply:
this part is from my old code;
this is referring to the IT pixel size here. The PS "P" side is in `zGeom` below